src/data.rs

```diff
@@ -570,7 +578,7 @@ impl<'a> IntoData<'a> for &'a mut Vec<u8> {
     type Output = Data<'a>;
 
     fn into_data(self) -> Result<Data<'a>> {
-        Data::from_seekable_stream(Cursor::new(self)).map_err(|e| e.error())
+        self.as_mut_slice().into_data()
```

Comment:
This should be reverted. The `Write` implementation for `Cursor<Vec<u8>>` (and `Cursor<&mut Vec<u8>>`) will increase the `Vec`'s capacity as needed which would not happen for just a slice.
```suggestion
        Data::from_seekable_stream(Cursor::new(self)).map_err(|e| e.error())
```